knip.jsonc

```diff
@@ -0,0 +1,46 @@
+{
+    "workspaces": {
+        ".": {
+            // These entries are complementary to the ones found in package.json
+            "entry": [
+                "lib/rules/index.js",
+                "tools/internal-rules/index.js",
+                "tools/update-rule-types.js",
+                "wdio.conf.js" // Remove when Knip has a wdio plugin
+            ],
+            "project": ["{conf,lib,tools}/**/*.js"],
+            "mocha": {
+                "entry": [
+                    "tests/{bin,conf,lib,tools}/**/*.js", // see Makefile.js
+                    "tests/_utils/test-lazy-loading-rules.js"
+                ],
+                "project": ["tests/**/*.js"]
+            },
+            "ignore": [
+                // If Knip would consider exports as named, their usage is too dynamic: globals[`es${ecmaVersion}`]
+                // An alternative is to add `__esModule: true` to the export and we can remove it here from the ignores:
+                "conf/globals.js",
```

Comment:
[Question] I don't understand 😞 could you elaborate please?

Reply:
- Knip is unable to infer from `globals[``es${ecmaVersion}``]` what exports are used from `conf/globals.js` (by `lib/source-code/source-code.js`).
- The alternative is to (cheat a little and) add `__esModule: true` (or anything that's not a shorthand property assignment) that effectively turns this export in a default export, resulting in that default export being imported by `lib/source-code/source-code.js`.